rest-api/offline/table_exporter.py

```diff
@@ -1,33 +1,114 @@
+import hashlib
+import random
 import re
+import struct
 
+from dao.database_factory import get_database
 from google.appengine.api import app_identity
 from google.appengine.ext import deferred
 from offline.sql_exporter import SqlExporter
 from werkzeug.exceptions import BadRequest
 
 _TABLE_PATTERN = re.compile("^[A-Za-z0-9_]+$")
 
+# TODO(calbach): Factor this out into the datastore config.
+_DEIDENTIFY_DB_TABLE_WHITELIST = {
+  'rdr': set([
+      'participant_view',
+      'physical_measurements_view',
+      'questionnaire_response_answer_view'
+  ])
+}
+
+
 class TableExporter(object):
   """API that exports data from our database to UTF-8 CSV files in GCS.
 
   Used instead of Cloud SQL export because it handles newlines and null characters in a way that
   other CSV clients (e.g. BigQuery, Google Sheets) can actually understand.
   """
 
+  @staticmethod
+  def _obfuscate_id(pmi_id, salt):
+    """
+    One-way ID obfuscation, intended for PMI participant IDs.
+
+    This method should aim to (1) avoid hash collisions across PMI ID inputs and (2) not be
+    reversible or rerunnable by anyone without RDR production data access.
+
+    Params:
+      - pmi_id (number) must be an integer participant ID
+      - salt (number) must be a long
+
+    Returns: a positive integer
+    """
+    h = hashlib.sha1()
+    h.update(struct.pack('>l', pmi_id))
+    h.update(salt)
+    # Just take the first 4 bytes so that the output ID is an integer, roughly in the same domain as
+    # the input PMI participant ID.
+    b = h.digest()[0:4]
```

Comment:
According to this:

https://stackoverflow.com/questions/1867191/probability-of-sha1-collisions

With 1M participants and 4 bytes, I think you're almost certainly going to have some collisions.

I'd recommend using 8 bytes instead?

There may be other ways of generating unique random IDs, but I guess they are going to be more complicated than this...